Firestore/core/src/firebase/firestore/api/document_reference.mm

```diff
@@ -191,31 +201,26 @@
                 // We don't raise `has_pending_writes` for deleted documents.
                 : false;
 
-        FIRDocumentSnapshot* result =
-            [FIRDocumentSnapshot snapshotWithFirestore:firestore
-                                           documentKey:key
-                                              document:document
-                                             fromCache:snapshot.from_cache()
-                                      hasPendingWrites:has_pending_writes];
-        listener(result, nil);
+        DocumentSnapshot result{firestore, key, document, snapshot.from_cache(),
+                                has_pending_writes};
+        listener(std::move(result));
       };
 
   FSTAsyncQueryListener* async_listener = [[FSTAsyncQueryListener alloc]
-      initWithExecutor:firestore_.client.userExecutor
+      initWithExecutor:firestore_->client().userExecutor
        snapshotHandler:std::move(handler)];
 
-  FSTQueryListener* internal_listener =
-      [firestore.client listenToQuery:query
-                              options:options
-                  viewSnapshotHandler:[async_listener asyncSnapshotHandler]];
-  return [[FSTListenerRegistration alloc] initWithClient:firestore_.client
+  FSTQueryListener* internal_listener = [firestore_->client()
+            listenToQuery:query
+                  options:options
+      viewSnapshotHandler:[async_listener asyncSnapshotHandler]];
+  return [[FSTListenerRegistration alloc] initWithClient:firestore_->client()
                                            asyncListener:async_listener
                                         internalListener:internal_listener];
 }
 
 bool operator==(const DocumentReference& lhs, const DocumentReference& rhs) {
-  return objc::Equals(lhs.firestore(), rhs.firestore()) &&
-         lhs.key() == rhs.key();
+  return lhs.firestore() == rhs.firestore() && lhs.key() == rhs.key();
```

Comment:
I _think_ comparing just the address is sufficient here, let me know what you think.

Reply:
SGTM